Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -1693,6 +1693,15 @@ void AccessibilityNodeObject::alterRangeValue(StepAction stepAction)
     if (!element || element->isDisabledFormControl())
         return;
 
+    // First try a keyboard event to see if that affects the value of the spinbutton, since authors are supposed to handle up/down keys.
+    // We can't check the event listeners directly here since those may be on an ancestor or the document.
+    if (role() == AccessibilityRole::SpinButton) {
+        float beforeValue = valueForRange();
+        postKeyboardKeysForValueChange(stepAction);
+        if (beforeValue != valueForRange())
+            return;
```

Comment:
Could be nice to have a comment explaining why we return if the value changed? I feel like it's obvious considering our recent discussions, but may not be obvious 5 years from now